dist/jspdf.debug.js

```diff
@@ -1130,9 +1130,14 @@
             arr.set(new Uint8Array([120, 156]));
             arr.set(p, 2);
             arr.set(new Uint8Array([adler32 & 0xff, adler32 >> 8 & 0xff, adler32 >> 16 & 0xff, adler32 >> 24 & 0xff]), p.byteLength + 2);
-            p = arr.reduce(function (data, byte) {
-              return data + String.fromCharCode(byte);
-            }, '');
+            var strings = [],
+                chunkSize = 0xffff; // There is a maximum stack size. We cannot call String.fromCharCode with as many arguments as we want
+
+            for (var j = 0; j * chunkSize < arr.length; j++) {
+              strings.push(String.fromCharCode.apply(null, arr.subarray(j * chunkSize, (j + 1) * chunkSize)));
```

Comment:
Cool - thank you! 

Some thoughts:

We could be slightly less aggressive here - I haven't tested and I don't know how browsers will react, here. IMHO "maximum stack size" should not only be affected by the arguments to the last method, but also to the calling stack - if the calling stack is deep, (which we have no control over) could this become a problem again? Haven't tested, but I guess we could also just reduce the chunksize to 0x8ffff and hope for the best ;-)   (for reference: https://stackoverflow.com/a/22747272/351836 )



I would have added a comment to explain that the upper bound of the subarray function is automatically clamped to valid values. I was not aware of this feature and thought that the code will fail in some cases.  

// end is clamped to valid array indices by "subarray"

Reply:
I have tried latest version of Chrome/Firefox/Edge and 0xffff works. I will test Safari on wednesday. For sure we can reduce amount of arguments to 0x8fff, if it's more safe...